pulsar/internal/connection_pool.go

```diff
@@ -65,7 +65,7 @@ func (p *connectionPool) GetConnection(logicalAddr *url.URL, physicalAddr *url.U
 			return cnx, nil
 		}
 		// The cached connection is failed
-		p.pool.Delete(logicalAddr.Host)
+		p.pool.Delete(fmt.Sprintf("%s:%v", logicalAddr.Host, connectingThroughProxy))
```

Comment:
This was a problem in the #146. It's being fixed in #161 .

The fix is not relevant in this PR which is changing a different aspect.

Reply:
Yes, will close the change.

Reply:
@wolfstudy The retry logic in the RPC client is good. We should add it to avoid pushing spurious errors back to the application (eg: when only 1 broker is getting restarted).